### DISCUSSION TITLE
Automatic reloading of extensions and server setup

### DISCUSSION THREAD

**Discussion by ArtursTimofejevs** [directus/directus#10436] [Feature Requests]
Hi :)

I thought I'd share something that has helped me develop extensions faster.

If you go to the server source folder and create a new folder called `extensions-develop` and have all your extensions source will live in that folder. Then you can use `nodemon` and `watch` to have automatic reloading of the extension and server whenever you save a file.

### Setup extension
* Create your folder where you will store your extensions source code, I put it in the source of the directus server
![image](https://user-images.githubusercontent.com/87968958/145570754-65ab5695-3123-47d1-8cd3-c715585df0d1.png)
* Initiate a new extension using the `npm init directus-extension` command inside of our new folder
* Add `--watch` flag in the extension package.json build command  
```
"scripts": {
  "build": "directus-extension build --watch"
},
```
* While still in `package.json` edit `directus:extensions.path` variable to point to the new location
```
"directus:extension": {
    "type": "endpoint",
    "path": "../../extensions/endpoints/cool-endpoint/index.js",
    "source": "src/index.ts",
    "host": "^9.2.1"
  },
```
* execute command line `npm run build` will now build to that folder each time you edit the extension

### Setup directus with nodemon
* install nodemon via `npm install nodemon`
* Edit package.json of directus (not the extension) in the root of the project
* Add the following command to scripts in package.json
`"start": "nodemon --exec npx directus start"`
* Add another file to the root of the directus project called `nodemon.json` with the following contents
```
{
  "watch": ["extensions/"]
}
```
* Now open a second command line and run `npm run start`

This will start the directus server so whenever you edit the extension source file the nodemon will detect a new build (or file change) in the extensions folder and reload the server.

Hope it helps :)

**Comment by rijkvanzanten** [comment]
Awesome! Thanks for sharing this knowledge @ArtursTimofejevs 🙂

**Reply by benhaynes** [reply]
Not technical enough to know if we can leverage this, but I know (from experience) that adding hot-reloading to extension dev would be amazing. 🙂

**Comment by jamescammarano** [comment]
A method to allow hot reloading was added in #10881 🥳 🎉

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
